packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -5116,9 +5117,11 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin, Res
             autofocus: true,
             child: UnmanagedRestorationScope(
               bucket: bucket,
-              child: Overlay(
-                key: _overlayKey,
-                initialEntries: overlay == null ?  _allRouteOverlayEntries.toList(growable: false) : const <OverlayEntry>[],
+              child: DefaultScrollAction(
```

Comment:
Also, wouldn't route A and B both have separate defaultScrollables? When route B is popped, how does Route A become the default again (I don't think there's any guarantee that the Scrollable's `didChangeDependencies` of Route A will run to re-register when Route B is popped.